phoenix-core/src/it/java/org/apache/phoenix/end2end/DropTableWithViewsIT.java

```diff
@@ -0,0 +1,164 @@
+            String ddlFormat =
+                    "CREATE TABLE IF NOT EXISTS " + baseTable + "  ("
+                            + " %s PK2 VARCHAR NOT NULL, V1 VARCHAR, V2 VARCHAR "
+                            + " CONSTRAINT NAME_PK PRIMARY KEY (%s PK2)" + " ) %s";
+            conn.createStatement().execute(generateDDL(ddlFormat));
+            conn.commit();
+
+            // Create a view tree (i.e., tree of views) with depth of 2 and fanout factor of 4
+            for (int  i = 0; i < 4; i++) {
+                String childView = SchemaUtil.getTableName(SCHEMA2, generateUniqueName());
+                String childViewDDL = "CREATE VIEW " + childView + " AS SELECT * FROM " + baseTable;
+                viewConn.createStatement().execute(childViewDDL);
+                for (int j = 0; j < 4; j++) {
+                    String grandChildView = SchemaUtil.getTableName(SCHEMA2, generateUniqueName());
+                    String grandChildViewDDL = "CREATE VIEW " + grandChildView + " AS SELECT * FROM " + childView;
+                    viewConn.createStatement().execute(grandChildViewDDL);
+                }
+            }
+            viewConn.commit();
+
+            // Drop the base table
+            String dropTable = String.format("DROP TABLE IF EXISTS %s CASCADE", baseTable);
+            conn.createStatement().execute(dropTable);
+            conn.commit();
+
+            try {
+                PhoenixRuntime.getTableNoCache(viewConn, baseTable.toUpperCase());
+                fail("An exception was not raised! the deleted table is loaded successfully");
+            }
+            catch (Throwable t) {
+                // As expected a Throwable exception happened
+            }
+
+            // Most of the views should be still available immediately after table drop
+            TableViewFinderResult childViewsResultBefore = new TableViewFinderResult();
+            TableName linkTable = TableName.valueOf(PhoenixDatabaseMetaData.SYSTEM_CHILD_LINK_NAME_BYTES);
+            ViewFinder.findAllRelatives(getUtility().getConnection().getTable(linkTable),
+                    HConstants.EMPTY_BYTE_ARRAY,
+                    SchemaUtil.getSchemaNameFromFullName(baseTable).getBytes(),
+                    SchemaUtil.getTableNameFromFullName(baseTable).getBytes(),
+                    PTable.LinkType.CHILD_TABLE,
+                    childViewsResultBefore);
+            assertTrue(childViewsResultBefore.getLinks().size() > 0);
+
+            // Wait for the tasks for dropping child views to complete. The depth of the view tree is 2, so we should wait
+            // 3 times the task handling interval by assuming that each non-root level will be processed in one interval
+
+            long timeInterval = config.getLong(
+                            QueryServices.TASK_HANDLING_INTERVAL_ATTRIB,
+                            QueryServicesOptions.DEFAULT_TASK_HANDLING_INTERVAL);
+
+            Thread.sleep(3*timeInterval);
```

Comment:
This might flap in some environments, can you run the task manually here so that the child views get cleaned up?